calculator.rb

```diff
@@ -0,0 +1,83 @@
+
+puts "*** Welcome to Natalie's Calculator! ***
+1. add(+)
+2. subtract(-)
+3. multiply(*)
+4. divide(/)
+-----------------
+Please chose an operator (name or symbol):"
+operator = gets.chomp
+
+# an array with all possible operators
+operators = ["+", "add", "-", "subtract", "*", "multiply", "times", "/", "divide"]
+
+until operators.include?(operator)
+  puts "That's not a valid operator. Select from the following list:
+  1. add(+)
+  2. subtract(-)
+  3. multiply(*)
+  4. divide(/)"
+  operator = gets.chomp
+end
+
+# The program handles erroneous input. For example the user might enter clown when asked to enter a number
+def is_string(a)
```

Comment:
This deals with almost all invalid user input. However, it does not recognize 0.0 as a valid number (there may be other input it mishandles). Can you think of a way to make this more robust?